fakeredis/_server.py

```diff
@@ -2422,6 +2443,11 @@ def __init__(self, server, db=0, password=None,
         self._parser = _DummyParser()
         self._sock = None
 
+    def connect(self):
+        super(FakeConnection, self).connect()
+        # The selector is set in redis.Connection.connect() after _connect() is called
+        self._selector = FakeSelector(self._sock)
```

Comment:
I'm not totally comfortable with letting the base class build a DefaultSelector with a fileno that might not even be valid, and is most likely not a socket. It works for now on Linux, but I'm not sure how portable it is, and if redis changes the default selector implementation (e.g. to use epoll) then I imagine it could break. I'm not sure what the best fix is though - possibly monkey-patching DefaultSelector so that it will return a FakeSelector when given a FakeSocket, but monkey-patching also tends to be quite fragile.

Any thoughts?

Reply:
I agree, this is not ideal. A real selector will be created on our fake socket which either uses `select.select` or `select.poll` The selector module does a good job of ensuring compatibility however and `0` is a valid socket on any system.

Reply:
`0` isn't normally a socket, but it is a file handle (stdin). I suspect that Windows won't be too happy (it doesn't treat sockets and file handles as interchangeably as POSIX does), but I'm not sure if redis-py runs on Windows anyway (neither I nor my colleagues use Windows, so I don't have a way to check).

It's also perfectly possible for a process to close FD 0 (I think some batch jobs do to detach from the terminal), or even to open a new file-like thing in their place. The good news is that I did a test with this and `select.poll` didn't seem to mind. I'd say let's just add some comments and move on, and fix it if it becomes a problem later.

Reply:
Ah, I see you added the comments already.